File: tests/machine/x/rb/match_full.rb
def classify(n)
	return (begin
	_t0 = n
	case
	when _t0 == 0
		"zero"
	when _t0 == 1
		"one"
	else
		"many"
	end
end)
end

$x = 2
$label = (begin
	_t1 = $x
	case
	when _t1 == 1
		"one"
	when _t1 == 2
		"two"
	when _t1 == 3
		"three"
	else
		"unknown"
	end
end)
puts($label)
$day = "sun"
$mood = (begin
	_t2 = $day
	case
	when _t2 == "mon"
		"tired"
	when _t2 == "fri"
		"excited"
	when _t2 == "sun"
		"relaxed"
	else
		"normal"
	end
end)
puts($mood)
$ok = true
$status = (begin
	_t3 = $ok
	case
	when _t3 == true
		"confirmed"
	when _t3 == false
		"denied"
	else
		nil
	end
end)
puts($status)
puts(classify(0))
puts(classify(5))

File: tests/machine/x/swift/match_full.swift
let x = 2
let label = { () in
    let __t = x
    switch __t {
    case 1: return "one"
    case 2: return "two"
    case 3: return "three"
    default: return "unknown"
    }
}()
print(label)
let day = "sun"
let mood = { () in
    let __t = day
    switch __t {
    case "mon": return "tired"
    case "fri": return "excited"
    case "sun": return "relaxed"
    default: return "normal"
    }
}()
print(mood)
let ok = true
let status = { () in
    let __t = ok
    switch __t {
    case true: return "confirmed"
    case false: return "denied"
    }
}()
print(status)
func classify(_ n: Int) -> String {
    return { () in
    let __t = n
    switch __t {
    case 0: return "zero"
    case 1: return "one"
    default: return "many"
    }
}()
}
print(classify(0))
print(classify(5))
